### PR TITLE
[BigQuery] PagedAPI Frontend support

### PR DESCRIPTION
**Pull request by MRzNone**

### CODE REVIEW

**Comment by MRzNone** [comment]
Functionally depends on [backend support](https://github.com/GoogleCloudPlatform/jupyter-extensions/pull/86).  Code will run, but no response.  Safe to deploy.

File: jupyterlab_bigquery/src/utils/pagedAPI/cache_manager.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+class CacheManager {
```
Comment (MRzNone) [inline jupyterlab_bigquery/src/utils/pagedAPI/cache_manager.ts:2]: Will add implementation in future PR.  Added as placehold for structure consistency.

Comment (MRzNone) [reply jupyterlab_bigquery/src/utils/pagedAPI/cache_manager.ts:2]: Decided not to implement cache.  Will delete in a future PR.

File: jupyterlab_bigquery/src/utils/widgetManager/widget_manager.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
+import rootReducer from '../../reducers';
+import { Widget } from '@phosphor/widgets';
+import { ReduxReactWidget } from './redux_react_widget';
+
+/**
+ * A class that manages dataset widget instances in the Main area
+ */
+export class WidgetManager {
```
Comment (cbwilkes) [inline jupyterlab_bigquery/src/utils/widgetManager/widget_manager.ts:11]: Can you sync this with the upstream? This file was refactored in https://github.com/GoogleCloudPlatform/jupyter-extensions/pull/83

Comment (MRzNone) [reply jupyterlab_bigquery/src/utils/widgetManager/widget_manager.ts:11]: Done

File: jupyterlab_bigquery/src/utils/pagedAPI/cache_manager.ts (hunk 1 above)
Comment (MRzNone) [reply jupyterlab_bigquery/src/utils/pagedAPI/cache_manager.ts]: In merging with master, I deleted cachemanager along the way.

File: jupyterlab_bigquery/src/components/list_items_panel/list_tree_item_widget.tsx [hunk 3]
```diff
@@ -1,16 +1,10 @@
...
-import { ReduxReactWidget } from '../../utils/ReduxReactWidget';
+import { ReduxReactWidget } from '../../utils/widgetManager/redux_react_widget';
 import { ListProjectsService } from './service/list_items';
-import { WidgetManager } from '../../utils/widget_manager';
 import ListItemsPanel from './list_tree_panel';
-
-export interface Context {
-  app: JupyterFrontEnd;
-  manager: WidgetManager;
-}
```
Comment (MRzNone) [inline jupyterlab_bigquery/src/components/list_items_panel/list_tree_item_widget.tsx:13]: Duplicate with [this in `list_tree_panel`](https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/6c915a28bba6e6f571922de585b90fc39c973710/jupyterlab_bigquery/src/components/list_items_panel/list_tree_panel.tsx#L21-L23).  Deleting here and rerouted all usage to that in `list_tree_panel`

**Pull request merged**